service/external_paysession.go

```diff
@@ -4,43 +4,68 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/http"
-
-	"github.com/companieshouse/payments.api.ch.gov.uk/models"
+	"strconv"
+	"strings"
 
 	"github.com/companieshouse/chs.go/log"
+	"github.com/companieshouse/payments.api.ch.gov.uk/models"
 )
 
 // CreateExternalPaymentJourney creates an external payment session with a Payment Provider that is given, e.g: GovPay
-func CreateExternalPaymentJourney(w http.ResponseWriter, req *http.Request) {
-	if req.Body == nil {
-		log.ErrorR(req, fmt.Errorf("request body empty"))
+func (service *PaymentService) CreateExternalPaymentJourney(w http.ResponseWriter, req *http.Request) {
+	id := req.URL.Query().Get(":payment_id")
+	if id == "" {
+		log.ErrorR(req, fmt.Errorf("payment id not supplied"))
 		w.WriteHeader(http.StatusBadRequest)
 		return
 	}
 
-	requestDecoder := json.NewDecoder(req.Body)
-	var incomingExternalPaymentJourneyRequest models.IncomingExternalPaymentJourneyRequest
-	err := requestDecoder.Decode(&incomingExternalPaymentJourneyRequest)
+	paymentSession, httpStatus, err := (*PaymentService).getPaymentSession(service, id)
```

Comment:
Is there any reason why the method has to be called this way? isn't it more readable to call `service.getPaymentSession(id)` instead?

Reply:
There wasn't any reason. Changed now.

Reply:
👍